crates/cli/build.rs

```diff
@@ -0,0 +1,59 @@
+use chrono::prelude::*;
+use std::fs;
+use std::process::Command;
+use std::str;
+
+fn main() {
+    // Rebuild if this build.rs file changes
+    println!("cargo:rerun-if-changed=build.rs");
+
+    // The version file is located at the root of the repository
+    let version_file_path = "../../version.txt";
+
+    // Rebuild if version file changes
+    println!("cargo:rerun-if-changed={}", version_file_path);
+
+    // Read the version file
+    let version_file_contents = fs::read_to_string(version_file_path).unwrap();
+
+    // If the version is "built-from-source", replace it with the git commit information
+    let version = match version_file_contents.trim() {
+        "built-from-source" => {
+            // Rebuild if a new Git commit is made
+            println!("cargo:rerun-if-changed=.git/HEAD");
+
+            // Get the hash of the current commit
+            let git_describe_output = Command::new("git")
+                .arg("describe")
+                .arg("--always")
+                .arg("--dirty= with changes") // Add a suffix if the working directory is dirty
+                .output()
+                .expect("Failed to execute git describe command");
+            let git_commit_hash = str::from_utf8(&git_describe_output.stdout)
+                .expect("Failed to parse git describe output")
+                .trim();
+
+            // Get the datetime of the last commit
+            let git_show_output = Command::new("git")
+                .arg("show")
+                .arg("--no-patch")
+                .arg("--format=%ct") // Outputting a UNIX timestamp is the only way to always use UTC
```

Comment:
I tried emitting a datetime in UTC, but all the ways I tried ended up giving me a local timezone. I don't want the compiler output to depend on the builder's timezone, so I get a UNIX timestamp from git and then parse it in Rust. Let me know if you can think of a better way to do this 🤷